internal/wrapper/githubsearchapi.go

```diff
@@ -0,0 +1,81 @@
+package wrapper
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+var (
+	SearchApiBaseUrl = "https://api.github.com/search"
+)
+
+type GithubSearchCodeResult struct {
+	FileURL string
+	Repo    string
+}
+
+type GithubSearchApiClient struct {
+	Token   string
+	Client  *http.Client
+	BaseURL string
```

Comment:
My advice is to never make anything public until you absolutely need to.
```suggestion
	token   string
	client  *http.Client
	baseURL string
```